### ISSUE TITLE
useFetch runaway effect not caught

### ISSUE THREAD

**Issue by joglr** [kentcdodds/stop-runaway-react-effects#7]
- `stop-runaway-react-effects` version: 1.2.0
- `node` version: 10.15.3
- `npm` (or `yarn`) version: 6.9.0

Relevant code or config
```javascript

function useGetProjects() {
  const projects = useFetch(endpoint, {
    method: 'post',
    body: JSON.stringify({ Token })
  })
  return projects
}

// useFetch
import { useEffect, useState } from 'react'

function useFetch(path, options) {
  const [result, setResult] = useState(null)
  useEffect(() => {
    let shouldCancel = false
    async function fetchData(path, options) {
      const response = await fetch(path, options)
      const result = await response.json()
      if (!shouldCancel) {
        setResult(result)
      }
    }
    fetchData(path, options)
    return () => {
      shouldCancel = true
    }
  }, [path, options])
                 // ^should be memoized, causes runaway setState calls
  return result
}

```

Since the options object passed to `useFetch` is recreated every call and not memoized, this caused a runaway effect. I tried using this module to catch future accidents like this, but it does not seem to catch it.

What you did:
 - Created `bootstrap-hijack.js` file from documentation
 - Imported it in the root file (index.js) 
 - Tried importing in the relevant file (App.jsx)

What happened:
 - The runaway hook was not caught

Reproduction repository:

I will try to get a repro repo together another time.

Problem description:

Suggested solution:

**Comment by kentcdodds** [comment]
Yeah, this is tricky. You can [customize the `callCount` and `timeLimit`](https://github.com/kentcdodds/stop-runaway-react-effects#api) if you like. But the really tricky thing is it's just hard to determine whether something's a "runaway" or if it's supposed to be functioning the way that it is. I'm not sure how best to determine that (which is why this probably isn't built-into React).

**Comment by joglr** [comment]
Thanks, I will try that when I find some more time to test it. 

It's a neat idea though, that potentially can save you from accidentally DDoS'ing your endpoints...

**Comment by tannerlinsley** [comment]
I ran into something similar today where an effect that relies on an animation loop was flagged as runaway. Like @kentcdodds said, there is not a very reliable way to determine if a runaway effect is false positive. What I would like though is some way to manually disable the runaway detection per effect.

**Comment by tbjgolden** [comment]
This eslint rule ("react-hooks/exhaustive-deps": "warn") may help you prevent this kind of thing happening:

https://github.com/facebook/react/blob/master/packages/eslint-plugin-react-hooks/src/ExhaustiveDeps.js#L516

**Comment by joglr** [comment]
> This eslint rule ("react-hooks/exhaustive-deps": "warn") may help you prevent this kind of thing happening:
> 
> https://github.com/facebook/react/blob/master/packages/eslint-plugin-react-hooks/src/ExhaustiveDeps.js#L516

This rule actually does the opposite for me, it causes these sorts of things, like infinite setState loop. That's probably because of some other problem though.
